specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/preview/2018-02-01-preview/imagebuilder.json

```diff
@@ -0,0 +1,799 @@
+        },
+        "sha256Checksum": {
+          "type": "string",
+          "description": "SHA256 Checksum of the ISO image."
+        }
+      },
+      "required": [
+        "sourceURI",
+        "sha256Checksum"
+      ]
+    },
+    "ImageTemplatePlatformImageSource": {
+      "description": "Describes an image source from [Azure Gallery Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages).",
+      "x-ms-discriminator-value": "PlatformImage",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ImageTemplateSource"
+        }
+      ],
+      "properties": {
+        "publisher": {
+          "type": "string",
+          "description": "Image Publisher in [Azure Gallery Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages)."
+        },
+        "offer": {
+          "type": "string",
+          "description": "Image offer from the [Azure Gallery Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages)."
+        },
+        "sku": {
+          "type": "string",
+          "description": "Image sku from the [Azure Gallery Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages)."
+        },
+        "version": {
+          "type": "string",
+          "description": "Image version from the [Azure Gallery Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages)."
+        }
+      }
+    },
+    "ImageTemplateCustomizer": {
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "The type of customization tool you want to use on the Image. For example, \"shell\" can be shellCustomizer"
+        },
+        "name": {
+          "type": "string",
+          "description": "Friendly Name to provide context on what this customization step does"
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "ImageTemplateShellCustomizer": {
+      "description": "Runs a shell script during the customization phase",
+      "x-ms-discriminator-value": "shell",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ImageTemplateCustomizer"
+        }
+      ],
+      "properties": {
+        "script": {
+          "type": "string",
+          "description": "The shell script to be run for customizing. It can be a github link, SAS URI for Azure Storage, etc"
+        }
+      }
+    },
+    "ImageTemplateDistributor": {
+      "type": "object",
+      "description": "Generic distribution object",
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Type of distribution."
+        },
+        "runOutputName": {
+          "type": "string",
+          "description": "The name to be used for the associated RunOutput.",
+          "pattern": "^[A-Za-z0-9-_]{1,64}$"
+        },
+        "tags": {
```

Comment:
why cant the top level "tags" property be used for this?

Reply:
I think Gaurav's question was more about naming being confusing with the usual resource "tags" property. Is it possible to change the name?

---
In reply to: [252806409](https://github.com/Azure/azure-rest-api-specs/pull/5055#discussion_r252806409) [](ancestors = 252806409)

Reply:
changed to 'artifactTags'